ec2/spark_ec2.py

```diff
@@ -423,13 +426,14 @@ def get_spark_ami(opts):
         b=opts.spark_ec2_git_branch)
 
     ami_path = "%s/%s/%s" % (ami_prefix, opts.region, instance_type)
+    reader = codecs.getreader("ascii")
     try:
-        ami = urlopen(ami_path).read().strip()
-        print("Spark AMI: " + ami)
+        ami = reader(urlopen(ami_path)).read().strip()
```

Comment:
Oh, interesting. I've generally seen people do this:

```
urlopen(ami_path).read().decode('ascii').strip()
```

Is there any material difference between doing a `decode()` like that versus using the reader like you've done?